python/src/dataset.rs

```diff
@@ -1367,15 +1367,34 @@ impl Dataset {
     }
 
     #[staticmethod]
-    #[pyo3(signature = (dest, storage_options = None))]
-    fn drop(dest: String, storage_options: Option<HashMap<String, String>>) -> PyResult<()> {
+    #[pyo3(signature = (dest, storage_options = None, ignore_not_found = true))]
```

Comment:
It looks like `ignore_not_found` is required here.  Can we either here to `Option<bool>` or change the `pyi` to `bool` instead of `Optional[bool]`?